ggml/src/ggml-cpu/ggml-cpu-quants.c

```diff
@@ -2381,10 +2384,11 @@ void ggml_vec_dot_q4_1_q8_1(int n, float * restrict s, size_t bs, const void * r
                 const int8x16_t y1_h = vld1q_s8(b_y1->qs + 16);
 
                 // mmla into int32x4_t
-                float32_t _scale[4] = {GGML_FP16_TO_FP32(b_x0->d)*b_y0->d,
-                    GGML_FP16_TO_FP32(b_x0->d)*b_y1->d,
-                    GGML_FP16_TO_FP32(b_x1->d)*b_y0->d,
-                    GGML_FP16_TO_FP32(b_x1->d)*b_y1->d};
+                float32_t _scale[4] = {
+                    GGML_FP16_TO_FP32(b_x0->d)*GGML_FP16_TO_FP32(b_y0->d),
+                    GGML_FP16_TO_FP32(b_x0->d)*GGML_FP16_TO_FP32(b_y1->d),
+                    GGML_FP16_TO_FP32(b_x1->d)*GGML_FP16_TO_FP32(b_y0->d),
+                    GGML_FP16_TO_FP32(b_x1->d)*GGML_FP16_TO_FP32(b_y1->d)};
```

Comment:
This fixes a bug where the `y->d` was not converted to F32, resulting in completely wrong numbers when running this branch.